client/src/ui/BitbakeCommands.ts

```diff
@@ -129,8 +129,8 @@ async function cleanRecipeCommand (bitbakeWorkspace: BitbakeWorkspace, bitbakeDr
   }
 }
 
-async function scanRecipeCommand (bitbakeWorkspace: BitbakeWorkspace, taskProvider: BitbakeTaskProvider, uri?: any): Promise<void> {
-  const chosenRecipe = await selectRecipe(bitbakeWorkspace, uri, false)
+async function scanRecipeCommand (bitbakeWorkspace: BitbakeWorkspace, taskProvider: BitbakeTaskProvider, bitBakeProjectScanner: BitBakeProjectScanner, uri?: any): Promise<void> {
+  const chosenRecipe = await selectRecipe(bitbakeWorkspace, bitBakeProjectScanner, uri, false)
```

Comment:
The command will accept recipes not in the workspace through the contextual menu. But you are right it doesn't make sense through the command pallet. I'll edit this specific case. Otherwise, we don't want it to flood the workspace.

We may need to scan results not in the workspace for some commands but not any planned currently. The new Workdir command will add it to the workspace though.

Reply:
Added a commit for this